src/Microsoft.SqlTools.ServiceLayer/Scripting/ScripterCore.cs

```diff
@@ -453,7 +446,71 @@ internal IEnumerable<Declaration> GetCompletionsForToken(ParseResult parseResult
             return Resolver.FindCompletions(
                 parseResult, parserLine, parserColumn, metadataDisplayInfoProvider);
         }
-        #endregion
 
+        /// <summary>
+        /// Wrapper method that calls Resolver.FindCompletions
+        /// </summary>
+        /// <param name="objectName"></param>
+        /// <param name="schemaName"></param>
+        /// <param name="objectType"></param>
+        /// <param name="tempFileName"></param>
+        /// <returns></returns>
+        internal ScriptingScriptOperation InitScriptOperation(string objectName, string schemaName, string objectType, string tempFileName)
+        {            
+            // object that has to be scripted
+            ScriptingObject obj = new ScriptingObject {
+                Name = objectName,
+                Schema = schemaName,
+                Type = objectType
+            };
+
+            // scripting options
+            ScriptOptions options = new ScriptOptions {
+			    ScriptCreateDrop = "ScriptCreate",
+			    TypeOfDataToScript = "SchemaOnly",
+			    ScriptStatistics = "ScriptStatsNone",
+			    TargetDatabaseEngineEdition = GetTargetDatabaseEngineEdition(),
+			    TargetDatabaseEngineType = GetTargetDatabaseEngineType(),
+			    ScriptCompatibilityOption = GetScriptCompatibilityOption(),
+                IncludeIfNotExists = true
+		    };
+
+            List<ScriptingObject> objList = new List<ScriptingObject>();
```

Comment:
>objList [](start = 34, length = 7)

we should prefer to not use shortened names list this.  Let's user objectList and includedList...assuming that is what the shortened names is supposed to mean.